package-testing/dotnet-sdk-relay/EppoSDKRelay/DTO/BanditActionRequest.cs

```diff
@@ -0,0 +1,29 @@
+using System.Text.Json;
+using eppo_sdk.dto.bandit;
+
+namespace EppoSDKRelay.DTO;
+
+public class BanditActionRequest
+{
+    public string Flag { get; set; }
+    public string DefaultValue { get; set; }
+    public string SubjectKey { get; set; }
+    public AttributeSet SubjectAttributes { get; set; }
+    public List<Action> Actions { get; set; }
+
+    public ContextAttributes GetSubjectContext()
+    {
+        return ContextAttributes.FromNullableAttributes(SubjectKey,
+                                                        SubjectAttributes.CategoricalAttributesAsStrings,
+                                                        SubjectAttributes.NumericAttributes);
+    }
+
+    public IDictionary<string, ContextAttributes> GetActionContextDict()
+    {
+        // The Eppo client API doesn't actually allow for non string-typed attributes to be explictly passed to the client.
+        // We get around these "dynamic" typing tests, not by skipping them like other languages, but by convering the data
```

Comment:
"converting"

Possibly dumb question: Is the reason we don't skip them because C# allows dynamic typing via `dynamic` so we want to make sure that can be input?

Reply:
good question.
This was actually from a deficiency in the attributes class types not having an explicit way of passing in a Dict<string, object> for the Categorical attributes.
Here, I've worked around by 1) converting from JsonValue objects to primitives and 2) using the `FromDict` method in `ContextAttributes` to automatically bucket based on type.

Reply:
also, to answer your question, we don't skip b/c we have the ability in c# to pass a string as `object` and get a string out the other side.